src/Driver/AbstractMySQLDriver.php

```diff
@@ -34,8 +35,14 @@ abstract class AbstractMySQLDriver implements VersionAwarePlatformDriver
     public function createDatabasePlatformForVersion($version)
     {
         $mariadb = stripos($version, 'mariadb') !== false;
-        if ($mariadb && version_compare($this->getMariaDbMysqlVersionNumber($version), '10.2.7', '>=')) {
-            return new MariaDb1027Platform();
+        if ($mariadb) {
+            if (version_compare($this->getMariaDbMysqlVersionNumber($version), '10.5.2', '>=')) {
+                return new MariaDb1052Platform();
+            }
+
+            if (version_compare($this->getMariaDbMysqlVersionNumber($version), '10.2.7', '>=')) {
```

Comment:
Let's do the version query only once.

```suggestion
            $mariaDbVersion = $this->getMariaDbMysqlVersionNumber($version);
            if (version_compare($mariaDbVersion, '10.5.2', '>=')) {
                return new MariaDb1052Platform();
            }

            if (version_compare($mariaDbVersion, '10.2.7', '>=')) {
```